SysBot.Pokemon.YouTube/Helpers/YouTubeTradeNotifier.cs

```diff
@@ -0,0 +1,96 @@
+
+            LogUtil.LogText($"Created trade details for {Username} - {Code}");
+        }
+
+        public Action<PokeRoutineExecutor> OnFinish { private get; set; }
+
+        public void SendNotification(PokeRoutineExecutor routine, PokeTradeDetail<T> info, string message)
+        {
+            LogUtil.LogText(message);
+            SendMessage($"@{info.Trainer.TrainerName}: {message}");
+        }
+
+        public void TradeCanceled(PokeRoutineExecutor routine, PokeTradeDetail<T> info, PokeTradeResult msg)
+        {
+            OnFinish?.Invoke(routine);
+            var line = $"@{info.Trainer.TrainerName}: Trade canceled, {msg}";
+            LogUtil.LogText(line);
+            SendMessage(line);
+        }
+
+        public void TradeFinished(PokeRoutineExecutor routine, PokeTradeDetail<T> info, T result)
+        {
+            OnFinish?.Invoke(routine);
+            var tradedToUser = Data.Species;
+            var message = $"@{info.Trainer.TrainerName}: " + (tradedToUser != 0 ? $"Trade finished. Enjoy your {(Species)tradedToUser}!" : "Trade finished!");
+            LogUtil.LogText(message);
+            SendMessage(message);
+        }
+
+        public void TradeInitialize(PokeRoutineExecutor routine, PokeTradeDetail<T> info)
+        {
+            var receive = Data.Species == 0 ? string.Empty : $" ({Data.Nickname})";
+            var msg = $"@{info.Trainer.TrainerName} (ID: {info.ID}): Initializing trade{receive} with you. Please be ready. Use the code you whispered me to search!";
+            msg += $" Your trade code is: {info.Code:0000 0000}";
+            LogUtil.LogText(msg);
+            SendMessage(msg);
+        }
+
+        public void TradeSearching(PokeRoutineExecutor routine, PokeTradeDetail<T> info)
+        {
+            var name = Info.TrainerName;
+            var trainer = string.IsNullOrEmpty(name) ? string.Empty : $", @{name}";
+            var message = $"I'm waiting for you{trainer}! My IGN is {routine.InGameName}.";
+            message += $" Your trade code is: {info.Code:0000 0000}";
+            LogUtil.LogText(message);
+            SendMessage($"@{info.Trainer.TrainerName} {message}");
+        }
+
+        public void SendNotification(PokeRoutineExecutor routine, PokeTradeDetail<T> info, PokeTradeSummary message)
+        {
+            var msg = message.Summary;
+            if (message.Details.Count > 0)
+                msg += ", " + string.Join(", ", message.Details.Select(z => $"{z.Heading}: {z.Detail}"));
+            LogUtil.LogText(msg);
+            SendMessage(msg);
+        }
+
+        public void SendNotification(PokeRoutineExecutor routine, PokeTradeDetail<T> info, T result, string message)
+        {
+            var msg = $"Details for {result.FileName}: " + message;
+            LogUtil.LogText(msg);
+            SendMessage(msg);
+        }
+
+        private void SendMessage(string message)
+        {
+            Client.SendMessage(message);
+        }
```

Comment:
Do we need this atall? as far as I can see, we dont use this function, so no point passing the client in the constructor either I think? Every other usage of SendMessage already has access to the client object